spec/integration/knife/config_use_spec.rb

```diff
@@ -67,6 +67,54 @@
       ENV["HOME"] = path_to(".")
     end
 
+    context "with no argument" do
+      context "with no configuration" do
+        it { is_expected.to eq "default\n" }
+      end
+
+      context "with --profile" do
+        let(:cmd_args) { %w{--profile production} }
+        it { is_expected.to eq "production\n" }
+      end
+
+      context "with an environment variable" do
+        around do |ex|
+          old_chef_profile = ENV["CHEF_PROFILE"]
+          begin
+            ENV["CHEF_PROFILE"] = "staging"
+            ex.run
+          ensure
+            ENV["CHEF_PROFILE"] = old_chef_profile
+          end
+        end
+
+        it { is_expected.to eq "staging\n" }
+      end
+
+      context "with a context file" do
+        before { file(".chef/context", "development\n") }
+        it { is_expected.to eq "development\n" }
+      end
+
+      context "with a context file under $CHEF_HOME" do
+        before do
+          file("chefhome/.chef/context", "other\n")
+          ENV["CHEF_HOME"] = path_to("chefhome")
+        end
+
+        it { is_expected.to eq "other\n" }
+      end
+
+      context "with a context file under $KNIFE_HOME" do
+        before do
+          file("knifehome/.chef/context", "other\n")
+          ENV["KNIFE_HOME"] = path_to("knifehome")
+        end
```

Comment:
Do these environment variables need to be reset after the test runs?

Reply:
Probably Yes, seems we are storing the old value of these environments variables and resetting back it to the original one.

Copy-paste stuff I haven't check in detail :smile: